Pixel.py

```diff
@@ -1,119 +1,314 @@
 from PIL import Image
 import math
-class Pixel:
+import JESConfig
 
-    def __init__(self, image, x, y):
+class Pixel:
+    """Provides access to pixels within a PIL image
+    """
+
+    def __init__(self, image=None, x=None, y=None):
+        """Pixel constructor
+
+        Parameters
+        ----------
+        image : PIL.Image
+            image the pixel belongs to
+        x : int
+            column of the pixel
+        y : int
+            row of hhe pixel
+        """
         self.image = image
         self.x = x
         self.y = y
         #self.color = color
 
-    #  * Method to return a string with information about this picture.
-    #  * @return a string with information about the picture such as fileName,
-    #  * height and width.
-    #  */
     def __str__(self):
-        pix = self.getImage().getpixel((self.x,self.y))
-        output = "Pixel red={} green={} blue={}".format(pix[0], pix[1], pix[2])
-        return output
+        """String for pixel
+
+        Returns
+        -------
+        str
+            user-readable pixel information
+        """
+        rgb = self.image.getpixel((self.x, self.y))
+        return "Pixel red={} green={} blue={}".format(rgb[0], rgb[1], rgb[2])
 
     def __repr__(self):
+        """Represetation of pixel
+
+        Returns
+        -------
+        str
+            string of pixel contents
+        """
         return self.__str__()
 
     def getX(self):
+        """Gets column containing pixel
+
+        Returns
+        -------
+        int
+           column containing the pixel
+        """
         return self.x
 
     def getY(self):
+        """Gets row containing pixel
+
+        Returns
+        -------
+        int
+           row containing the pixel
+        """
         return self.y
 
-    def getImage(self):
```

Comment:
Do we want to get rid of getImage()? I understand that locally we can use self.image, but wouldn't it be safer for files that import Pixel to have access to getImage rather than self.image?